pkg/endpoint/bpf.go

```diff
@@ -459,34 +459,53 @@ func (e *Endpoint) removeOldRedirects(owner Owner, desiredRedirects map[string]b
 		}
 }
 
-// regenerateBPF rewrites all headers and updates all BPF maps to reflect the
-// specified endpoint.
-// Must be called with endpoint.Mutex not held and endpoint.BuildMutex held.
-// Returns the policy revision number when the regeneration has called, a
-// boolean if the BPF compilation was executed and an error in case of an error.
-func (e *Endpoint) regenerateBPF(owner Owner, currentDir, nextDir string, regenContext *RegenerationContext) (revnum uint64, compiled bool, reterr error) {
-	var (
-		err                 error
-		compilationExecuted bool
-	)
+// datapathRegenerationContext contains information related to regenerating the
+// datapath (BPF, proxy, etc.).
+type datapathRegenerationContext struct {
```

Comment:
I wonder if this structure should be part of the RegenerationContext itself. Open to suggestions.

Reply:
Also am open to different names for these fields, etc.